src/main/msp/msp.c

```diff
@@ -2318,20 +2381,121 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
             if (sbufBytesRemaining(src) >= 2) {
                 vtxSettingsConfigMutable()->power = sbufReadU8(src);
+                const uint8_t newPitmode = sbufReadU8(src);
                 if (vtxType != VTXDEV_UNKNOWN) {
                     // Delegate pitmode to vtx directly
```

Comment:
The comment though is still in context of whether the `vtxType != VTXDEV_UNKNOWN` and pit mode can be sent. Moving the `sbufReadU8()` out of the condition is really a bug fix since before it would leave an unread byte in the buffer if the condition wasn't true. Didn't matter before because nothing followed, but matters now.